aws-s3-source/src/main/java/dev/knative/eventing/KnativeClientOptions.java

```diff
@@ -0,0 +1,60 @@
+package dev.knative.eventing;
+
+import io.quarkus.tls.runtime.keystores.TrustAllOptions;
+import io.vertx.core.net.JksOptions;
+import io.vertx.core.net.PemKeyCertOptions;
+import io.vertx.core.net.PfxOptions;
+import io.vertx.ext.web.client.WebClientOptions;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Named;
+import org.apache.camel.CamelContext;
+import org.apache.camel.spi.PropertiesComponent;
+
+/**
+ * Knative client options able to configure secure Http transport options.
+ * SSL options are added automatically when enabled via
+ */
+@ApplicationScoped
+public class KnativeClientOptions {
+
+    private static final String PROPERTY_PREFIX = "camel.knative.client.ssl.";
+
+    @Named("knativeClientOptions")
+    public WebClientOptions knativeClientOptions(CamelContext camelContext) {
+        PropertiesComponent propertiesComponent = camelContext.getPropertiesComponent();
+
+        boolean sslEnabled = Boolean.parseBoolean(
+                propertiesComponent.resolveProperty(PROPERTY_PREFIX + "enabled").orElse("false"));
+
+        WebClientOptions options = new WebClientOptions();
+        if (sslEnabled) {
+            options.setSsl(true);
+
+            boolean verifyHostname = Boolean.parseBoolean(
+                    propertiesComponent.resolveProperty(PROPERTY_PREFIX + "verify.hostname").orElse("false"));
+            options.setVerifyHost(verifyHostname);
+
+            String keystorePath = propertiesComponent.resolveProperty(PROPERTY_PREFIX + "keystore.path").orElse("");
+            String keystorePassword = propertiesComponent.resolveProperty(PROPERTY_PREFIX + "keystore.password").orElse("");
+
+            if (!keystorePath.isEmpty()) {
+                if (keystorePath.endsWith(".p12")) {
+                    options.setKeyCertOptions(new PfxOptions().setPath(keystorePath).setPassword(keystorePassword));
+                } else {
+                    options.setKeyCertOptions(new JksOptions().setPath(keystorePath).setPassword(keystorePassword));
+                }
+            }
+
+            String keyPath = propertiesComponent.resolveProperty(PROPERTY_PREFIX + "key.path").orElse("");
+            String certPath = propertiesComponent.resolveProperty(PROPERTY_PREFIX + "cert.path").orElse(keyPath);
+
+            if (!keyPath.isEmpty()) {
+                options.setKeyCertOptions(new PemKeyCertOptions().setKeyPath(keyPath).setCertPath(certPath));
+            }
+
+            options.setTrustOptions(TrustAllOptions.INSTANCE);
```

Comment:
is this config trusting any CA without using `/knative-custom-certs/...` config?

Reply:
So far I have only seen keystores being injected into `/knative-custom-certs/`. should `knative-eventing-bundle.jks` also be used as a truststore then?

I will add a configuration option for truststore path and password and just use TrustAll option as a default when nothing is set, WDYT?